src/Buffer.h

```diff
@@ -128,6 +128,9 @@ class Buffer {
     /** Trivial copy assignment operator. */
     Buffer &operator=(const Buffer &that) = default;
 
+    /** Trivial move assignment operator. */
```

Comment:
What does this do with the source object? (Yes, I should figure it out myself. If you are sure this doesn't drop the reference in destructing the temporary being assigned from, then all good.)

Reply:
A Buffer<> is just an IntrusivePtr, which is safe to move.